nighthawk/source/client/options_impl.cc

```diff
@@ -0,0 +1,141 @@
+                                     "The number of seconds that the test should run. Default: 5.",
+                                     false, 5, "uint64_t", cmd);
+  TCLAP::ValueArg<uint64_t> timeout(
+      "", "timeout",
+      "Timeout period in seconds used for both connection timeout and grace period waiting for "
+      "lagging responses to come in after the test run is done. Default: 5.",
+      false, 5, "uint64_t", cmd);
+
+  TCLAP::SwitchArg h2("", "h2", "Use HTTP/2", cmd);
+
+  TCLAP::ValueArg<std::string> concurrency(
+      "", "concurrency",
+      "The number of concurrent event loops that should be used. Specify 'auto' to let "
+      "Nighthawk leverage all vCPUs that have affinity to the Nighthawk process.Note that "
+      "increasing this results in an effective load multiplier combined with the configured-- rps "
+      "and --connections values.Default : 1. ",
+      false, "1", "string", cmd);
+
+  std::vector<std::string> log_levels;
+  log_levels.emplace_back("trace");
+  log_levels.emplace_back("debug");
+  log_levels.emplace_back("info");
+  log_levels.emplace_back("warn");
+  log_levels.emplace_back("error");
+  TCLAP::ValuesConstraint<std::string> verbosities_allowed(log_levels);
+
+  TCLAP::ValueArg<std::string> verbosity(
+      "v", "verbosity",
+      "Verbosity of the output. Possible values: [trace, debug, info, warn, error, critical]. The "
+      "default level is 'info'.",
+      false, "info", &verbosities_allowed, cmd);
+
+  TCLAP::UnlabeledValueArg<std::string> uri("uri",
+                                            "uri to benchmark. http:// and https:// are supported, "
+                                            "but in case of https no certificates are validated.",
+                                            true, "", "uri format", cmd);
+
+  cmd.setExceptionHandling(false);
+  try {
+    cmd.parse(argc, argv);
+  } catch (TCLAP::ArgException& e) {
+    try {
+      cmd.getOutput()->failure(cmd, e);
+    } catch (const TCLAP::ExitException&) {
+      // failure() has already written an informative message to stderr, so all that's left to do
+      // is throw our own exception with the original message.
+      throw MalformedArgvException(e.what());
+    }
+  } catch (const TCLAP::ExitException& e) {
+    // parse() throws an ExitException with status 0 after printing the output for --help and
+    // --version.
+    throw NoServingException();
+  }
+
+  requests_per_second_ = requests_per_second.getValue();
+  connections_ = connections.getValue();
+  duration_ = duration.getValue();
+  timeout_ = timeout.getValue();
+  uri_ = uri.getValue();
+  h2_ = h2.getValue();
+  concurrency_ = concurrency.getValue();
+  verbosity_ = verbosity.getValue();
+
+  uint64_t pretty_large_value = 1000 * 1000 * 1000;
+  // We cap on negative values. TCLAP accepts negative values which we will get here as very
+  // large values. We just cap values to a very large maximum.
```

Comment:
It seems we should either have per-option max (and put it in docs) or just allow anything that is representable in 2^63.